applications/ecallmgr/src/ecallmgr_fs_route.erl

```diff
@@ -347,6 +347,9 @@ start_message_handling(_Node, _FetchId, CallId, JObj) ->
 
 -spec route_req(ne_binary(), ne_binary(), wh_proplist(), atom()) -> wh_proplist().
 route_req(CallId, FetchId, Props, Node) ->
+    SwitchURL = ecallmgr_fs_node:sip_url(Node),
+    [_, SwitchURIHost] = binary:split(SwitchURL, <<"@">>),
```

Comment:
So should the handle_info/2 of ecallmgr_fs_event_stream, when it sets `switch_info` to `true` to process the event, get the return of the sub-call to handle_info and set `switch_info` back to `false`? Otherwise, if it is missed the first time, the switch info won't be available on that event stream.